tests/dump_tests/module_tests/copp_test.py

```diff
@@ -0,0 +1,197 @@
+import json, os, sys
+import jsonpatch
+import unittest
+import pytest
+from deepdiff import DeepDiff
+from mock import patch
+from dump.helper import display_template, sort_lists
+from dump.plugins.copp import Copp
+
+module_tests_path = os.path.dirname(__file__)
+dump_tests_path = os.path.join(module_tests_path, "../")
+copp_files_path = os.path.join(dump_tests_path,"files","copp")
+sys.path.append(dump_tests_path)
+sys.path.append(module_tests_path)
+
+from module_tests.mock_sonicv2connector import MockSonicV2Connector
+
+dedicated_dbs = {}
+dedicated_dbs['CONFIG_DB'] = os.path.join(copp_files_path, "config_db.json") 
+dedicated_dbs['APPL_DB'] = os.path.join(copp_files_path, "appl_db.json") 
+dedicated_dbs['ASIC_DB'] = os.path.join(copp_files_path, "asic_db.json")
+dedicated_dbs['STATE_DB'] = os.path.join(copp_files_path, "state_db.json")
+
+def mock_connector(host, namespace):
+    return MockSonicV2Connector(dedicated_dbs, namespace)
+
+@pytest.fixture(scope="module", autouse=True)
+def verbosity_setup():
+    print("SETUP")
+    os.environ["VERBOSE"] = "1"
+    yield
+    print("TEARDOWN")
+    os.environ["VERBOSE"] = "0"
+
+@patch("dump.match_infra.SonicV2Connector", mock_connector)
+@patch("dump.plugins.copp.Copp.CONFIG_FILE", os.path.join(dump_tests_path, "files/copp_cfg.json"))
+class TestCoppModule(unittest.TestCase):
+    
+    def test_custom_trap_with_default_grp(self):
+        params = {}
+        params[Copp.ARG_NAME] = "snmp"
+        params["namespace"] = ""
+        m_copp = Copp()
+        returned = m_copp.execute(params)
+        print(returned)
+        expect = display_template(dbs=["CONFIG_FILE", "CONFIG_DB", "APPL_DB", "ASIC_DB", "STATE_DB"])
+        expect["CONFIG_FILE"]["keys"].append("COPP_GROUP|queue4_group2")
+        expect["CONFIG_DB"]["keys"].append("COPP_TRAP|snmp_grp")
+        expect["APPL_DB"]["keys"].append("COPP_TABLE:queue4_group2")
+        expect["STATE_DB"]["keys"].extend(["COPP_GROUP_TABLE|queue4_group2", "COPP_TRAP_TABLE|snmp_grp"])
+        expect["ASIC_DB"]["keys"].extend(["ASIC_STATE:SAI_OBJECT_TYPE_HOSTIF_TRAP:oid:0x220000000004dc", "ASIC_STATE:SAI_OBJECT_TYPE_HOSTIF_TRAP_GROUP:oid:0x110000000004da", 
+                                          "ASIC_STATE:SAI_OBJECT_TYPE_POLICER:oid:0x120000000004db","ASIC_STATE:SAI_OBJECT_TYPE_QUEUE:oid:0x150000000002a0"])
+        ddiff = DeepDiff(sort_lists(returned), sort_lists(expect))
+        assert not ddiff, ddiff
+    
+    def test_default_trap_and_default_grp(self):
```

Comment:
Please rename as there is a "default" trap group in sonic and it might create confusion